klangbecken_api.py

```diff
@@ -71,21 +72,16 @@ def _full_path(self, path):
         return pjoin(self.data_dir, path)
 
     def _replaygain_analysis(self, mutagenfile):
-        # 1. compute track_gain
-        gstreamer_cmd = [
-            "gst-launch-1.0", "-t",
-            "filesrc", "location="+mutagenfile.filename,
-            "!", "decodebin",
-            "!", "audioconvert",
-            "!", "audioresample",
-            "!", "rganalysis",
-            "!", "fakesink"
+        bs1770gain_cmd = [
+            "/usr/bin/bs1770gain", "--ebu", "--xml", mutagenfile.filename
         ]
-        output = str(subprocess.check_output(gstreamer_cmd))
-        search_str = 'replaygain track gain: '
-        start = output.find(search_str) + len(search_str)
-        track_gain = output[start:None].split('\\n')[0]
-        # 2. set TXXX tag
+        output = subprocess.check_output(bs1770gain_cmd)
+        bs1770gain = ElementTree.fromstring(output)
+        # lu is in bs1770gain > album > track > integrated as an attribute
+        track_gain = bs1770gain.find('album') \
```

Comment:
I think you can also use xPath with find... Something along the (untested) lines of `find('album/track/integrated').attrib('lu')`.

Reply:
Ok I somehow overlooked in the docs that you can use XPath with ElementTree :+1: